### PR TITLE
Get all equipments when country criteria is null

### PR DESCRIPTION
**Pull request by etiennehomer**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
https://github.com/powsybl/powsybl-core/issues/2495
See 7/

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
When countries of SingleCountryCriterion is empty, no contingency is returned, all equipments are filtrered. 

**What is the new behavior (if this is a feature change)?**
When countries of SingleCountryCriterion is empty, all equipments are returned, the filter is transparent.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [x] The *Breaking Change*

### CODE REVIEW

File: contingency/contingency-api/src/main/java/com/powsybl/contingency/contingency/list/criterion/SingleCountryCriterion.java [hunk 1]
```diff
@@ -73,6 +73,6 @@ private boolean filterInjection(VoltageLevel voltageLevel) {
         if (injectionCountry == null) {
             return false;
         }
-        return countries.contains(injectionCountry);
+        return countries == null || countries.isEmpty() || countries.contains(injectionCountry);
```
[the remark this answers is not in the document]
Comment (EtienneLt) [reply contingency/contingency-api/src/main/java/com/powsybl/contingency/contingency/list/criterion/SingleCountryCriterion.java:76]: maybe just empty ?

File: contingency/contingency-api/src/test/resources/injectionCriterionContingencyListNoCountry.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{
+  "type" : "injectionCriterion",
+  "version" : "1.0",
+  "name" : "list2",
+  "identifiableType" : "GENERATOR",
+  "countryCriterion" : {
+    "type" : "SINGLE_COUNTRY",
+    "countries" : [ ]
+  },
```
Comment (EtienneLt) [inline contingency/contingency-api/src/test/resources/injectionCriterionContingencyListNoCountry.json:9]: if there is no country maybe we not add countryCriterion in the json

File: contingency/contingency-api/src/main/java/com/powsybl/contingency/contingency/list/criterion/SingleCountryCriterion.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (etiennehomer) [reply contingency/contingency-api/src/main/java/com/powsybl/contingency/contingency/list/criterion/SingleCountryCriterion.java]: Ok with that

File: contingency/contingency-api/src/main/java/com/powsybl/contingency/contingency/list/criterion/SingleCountryCriterion.java [hunk 3]
```diff
@@ -58,10 +58,10 @@ private boolean filterSubstation(Substation substation) {
             return false;
         }
         Country injectionCountry = substation.getCountry().orElse(null);
-        if (injectionCountry == null) {
+        if (injectionCountry == null && !countries.isEmpty()) {
```
Comment (miovd) [inline contingency/contingency-api/src/main/java/com/powsybl/contingency/contingency/list/criterion/SingleCountryCriterion.java:61]: From what I understand `countries` should not be `null`: add a `Objects.requireNonNull` in the constructor

File: contingency/contingency-api/src/main/java/com/powsybl/contingency/contingency/list/criterion/TwoCountriesCriterion.java [hunk 4]
```diff
@@ -50,7 +55,7 @@ private boolean filterBranch(Terminal terminal1, Terminal terminal2) {
         }
         Country countrySide1 = substation1.getCountry().orElse(null);
         Country countrySide2 = substation2.getCountry().orElse(null);
-        if (countrySide1 == null || countrySide2 == null) {
+        if (countrySide1 == null && !countries1.isEmpty() || countrySide2 == null && !countries2.isEmpty()) {
```
Comment (miovd) [inline contingency/contingency-api/src/main/java/com/powsybl/contingency/contingency/list/criterion/TwoCountriesCriterion.java:58]: Same thing: make sure in the constructor that the lists are not null

File: contingency/contingency-api/src/test/resources/injectionCriterionContingencyListNoCountry.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{
+  "type" : "injectionCriterion",
+  "version" : "1.0",
+  "name" : "list2",
+  "identifiableType" : "GENERATOR",
+  "countryCriterion" : null,
```
Comment (miovd) [inline contingency/contingency-api/src/test/resources/injectionCriterionContingencyListNoCountry.json:6]: I'm confused: why is null here and not in `injectionCriterionContingencyListEmptyCountryList`?

**Review by miovd — COMMENTED** [review]
Okay a question for me: In `Substations`, the convention was that if the `country` filter is `null`, all equipment must be returned, but if it is empty, it should filter.
In practice, I agree with what you did but did you do it (and indeed, an empty filter does not make a ton of sense) but this convention might be confusing: at least add a javadoc about it in `SingleCountryCriterion` and `TwoContriesCriterion` and ensure that the lists can never be null (it is a breaking change)

File: contingency/contingency-api/src/main/java/com/powsybl/contingency/contingency/list/criterion/SingleCountryCriterion.java (hunk 3 above)
Comment (etiennehomer) [reply contingency/contingency-api/src/main/java/com/powsybl/contingency/contingency/list/criterion/SingleCountryCriterion.java:61]: Done

File: contingency/contingency-api/src/main/java/com/powsybl/contingency/contingency/list/criterion/TwoCountriesCriterion.java (hunk 4 above)
Comment (etiennehomer) [reply contingency/contingency-api/src/main/java/com/powsybl/contingency/contingency/list/criterion/TwoCountriesCriterion.java:58]: Done

File: contingency/contingency-api/src/test/resources/injectionCriterionContingencyListNoCountry.json (hunk 5 above)
Comment (etiennehomer) [reply contingency/contingency-api/src/test/resources/injectionCriterionContingencyListNoCountry.json:6]: I changed the json test file name to make it more explicit: there's one test for an injectionCriterion that has no country criterion and one test for an injectionCriterion with a country criterion which its country list is an empty list.

**Pull request closed without merging**